src/ast/nodes/shared/BitFlags.ts

```diff
@@ -0,0 +1,41 @@
+export const enum Flag {
+	included = 0b0000_0000_0000_0000_0000_0000_0000_0001,
+	deoptimized = 0b0000_0000_0000_0000_0000_0000_0000_0010,
+	tdzAccessDefined = 0b0000_0000_0000_0000_0000_0000_0000_0100,
+	tdzAccess = 0b0000_0000_0000_0000_0000_0000_0000_1000,
+	assignmentDeoptimized = 0b0000_0000_0000_0000_0000_0000_0001_0000,
+	bound = 0b0000_0000_0000_0000_0000_0000_0010_0000,
+	isUndefined = 0b0000_0000_0000_0000_0000_0000_0100_0000,
+	optional = 0b0000_0000_0000_0000_0000_0000_1000_0000,
+	async = 0b0000_0000_0000_0000_0000_0001_0000_0000,
+	deoptimizedReturn = 0b0000_0000_0000_0000_0000_0010_0000_0000,
+	computed = 0b0000_0000_0000_0000_0000_0100_0000_0000,
+	hasLostTrack = 0b0000_0000_0000_0000_0000_1000_0000_0000,
+	hasUnknownDeoptimizedInteger = 0b0000_0000_0000_0000_0001_0000_0000_0000,
+	hasUnknownDeoptimizedProperty = 0b0000_0000_0000_0000_0010_0000_0000_0000,
+	directlyIncluded = 0b0000_0000_0000_0000_0100_0000_0000_0000,
+	deoptimizeBody = 0b0000_0000_0000_0000_1000_0000_0000_0000,
+	isBranchResolutionAnalysed = 0b0000_0000_0000_0001_0000_0000_0000_0000,
+	await = 0b0000_0000_0000_0010_0000_0000_0000_0000,
+	method = 0b0000_0000_0000_0100_0000_0000_0000_0000,
+	shorthand = 0b0000_0000_0000_1000_0000_0000_0000_0000,
+	tail = 0b0000_0000_0001_0000_0000_0000_0000_0000,
+	prefix = 0b0000_0000_0010_0000_0000_0000_0000_0000
+}
+
+export function setFlag(oldFlag: number, mask: Flag, value: boolean): number {
+	// This is a branchless version of the following code:
+	// return value ? (oldFlag | mask) : (oldFlag & ~mask);
+	//
+	// I profiled these variants, and there was essentially no difference in performance, however,
+	// the branchless version apparently is more likely to be inlined by v8 and the ternary version
+	// was giving me issues with maximum call stack size exceeded errors.
```

Comment:
This apparently is not true anymore after re-testing this. Its possible the code I was trying before was missing the final assignment to `this.flags` as that can also cause causes the same problem when the boolean sets don't get applied. I'm re-testing the performance with the set using a ternary and moving this from a free-function to a protected member function on `ExpressionEntity`. I didn't originally implement this as a member function because I was worried that I might need to make use of these functions outside of the AST node class hierarchy, but that didn't wind up being necessary at all.

Assuming performance pans out to be roughly the same, I will update the PR to make the following two changes:
1) Switch `setFlag` and `isFlagSet` to be member functions with this.setFlag mutating the flag directly instead of returning the new flag
2) switch the setFlag implementation to use the more clear ternary implementation since performance should be more or less identical.

Reply:
After investigating, I found that the member function was slower than the free function version in spite of the code being cleaner.